lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,222 @@
+//
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:flutter/material.dart';
+
+import '../../utils/sample_state_support.dart';
+
+class SetReferenceScaleSample extends StatefulWidget {
+  const SetReferenceScaleSample({super.key});
+
+  @override
+  State<SetReferenceScaleSample> createState() => _SetReferenceScaleState();
+}
+
+class _SetReferenceScaleState extends State<SetReferenceScaleSample>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // Create a list of dropdown menu items to load all the reference scales.
+  final _referenceScaleList = <DropdownMenuItem<double>>[];
+  // Create a list of selected feature layers.
+  final _selectedFeatureLayers = <String>[];
```

Comment:
This is looking good! The only thing that jumps out now is that the reference scale is being updated in all layers when I toggle initially despite them being all unchecked- so I think maybe we need to check the initial state. Maybe they can all start as selected - so we just need to ensure the checkboxes reflect this. I haven't had chance to look in detail at why, but maybe have a play around. Once I toggle the checkboxes on/off, it works as expected.

Reply:
Oh yes I think I know what you mean - I noticed they are all checked out for iOS sample so I will make sure to update that in onMapViewReady and have them display the initial state of the layer.refenceSymbol